crates/sui-graphql-rpc/src/types/dry_run_result.rs

```diff
@@ -92,21 +92,29 @@ impl TryFrom<SuiExecutionResult> for DryRunEffect {
 
 impl TryFrom<DevInspectResults> for DryRunResult {
     type Error = crate::error::Error;
-    fn try_from(results: DevInspectResults) -> Result<Self, Self::Error> {
-        let execution_results = results
-            .results
-            .ok_or_else(|| {
-                Error::Internal("No execution results returned from dev inspect".to_string())
-            })?
+    fn try_from(dev_inspect_results: DevInspectResults) -> Result<Self, Self::Error> {
+        // Results might be None in the event of a transaction failure.
+        let results = if let Some(results) = dev_inspect_results.results {
+            Some(
+                results
+                    .into_iter()
+                    .map(DryRunEffect::try_from)
+                    .collect::<Result<Vec<_>, Error>>()?,
```

Comment:
Will this return an error in case DryRun fails?

Reply:
When a dry run fails, `dev_inspect_results.results` would be `None` and `dev_inspect_results.error` would be `Some`. If the dry run succeeds, then the reverse is true. Here we are in the branch where the dry run succeeds and `results` is `Some`. And to clarify, the `Error` in  `collect::<Result<Vec<_>, Error>>()?` is potential internal errors from `DryRunEffect::try_from`.